.github/workflows/ci.yml

```diff
@@ -0,0 +1,70 @@
+name: Build & Tests
+
+on:
+  push:
+  pull_request:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build_test:
+    # Used for installing Miri
+    # https://rust-lang.github.io/rustup-components-history/
+    env:
+      MIRI_TOOLCHAIN_VERSION: '2022-09-26'
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        channel: [ "1.56.1", "stable", "beta", "nightly" ]
+        target: [ "i686-unknown-linux-gnu", "x86_64-unknown-linux-gnu", "arm-unknown-linux-gnueabi", "aarch64-unknown-linux-gnu", "powerpc-unknown-linux-gnu", "powerpc64-unknown-linux-gnu", "wasm32-wasi" ]
+        features: [ "" , "--features alloc,simd,simd-nightly"]
+
+    name: Build & Test (${{ matrix.channel }} for ${{ matrix.target }}, features set to ${{ matrix.features != '' }})
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Setup Rust ${{ matrix.channel }} ${{ matrix.target }}
+      uses: ATiltedTree/setup-rust@v1
+      with:
+        rust-version: ${{ matrix.channel }}
+        targets: ${{ matrix.target }}
+    
+    - name: Rust Cache
+      uses: Swatinem/rust-cache@v2.0.0
+      with:
+        key: ${{ matrix.channel }}-${{ matrix.target }}-${{ matrix.features == '' }}-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Won't `${{ matrix.features == '' }}` have the effect of treating all sets of features as identical? Would it make more sense to do `${{ matrix.features }}` so that the exact feature set is part of the key? If that causes syntax issues, you could update the `features` variable defined above to:

```text
features: [ "", "alloc,simd,simd-nightly" ]
```

...and then use it in the various cargo commands below as `--features ${{ matrix.features }}` (although you might need to put that in quotes in case it's the empty string - `"${{ matrix.features }}"` - I'm not sure whether the `run` command passes its input to the shell as a single string or as separate arguments).

Reply:
`${{ matrix.features == '' }}` will return a boolean and since the features vector is either empty or `--features alloc,simd,simd-nightly` it should be `[true, false]`

Reply:
Agreed, but we might want to update the `features` vector in the future (for example, to have a separate feature set for stable features: `[ "", "alloc,simd", "alloc,simd,simd-nightly" ]`).

Reply:
The issue with this is that we can't have commas in the key because that's what I tried initially.

Reply:
Ah understood. I guess let's merge as-is and I can try to change this after it's merged. I'll submit an issue for this once this is merged.